pandas/tests/io/test_pytables.py

```diff
@@ -4928,6 +4928,23 @@ def test_categorical_conversion(self):
             result = read_hdf(path, 'df', where='obsids=B')
             tm.assert_frame_equal(result, expected)
 
+    def test_categorical_nan_only_columns(self):
+        # GH18413
+        # Check that read_hdf with categorical columns with NaN-only values can
+        # be read back.
+        df = pd.DataFrame({
+            'a': ['a', 'b', 'c', np.nan],
+            'b': [np.nan, np.nan, np.nan, np.nan],
```

Comment:
can you add another column with an array like ``pd.Series([None]* 3, dtype=object)``. this might fail your test because the original array was an all-null object type (and no float). but let's see.

Reply:
done, seems to work fine.